### PR TITLE
Ability to attach watermark when upload images to gallery

### PR DESCRIPTION
**Pull request by pitchayakit**

### CODE REVIEW

File: src/Classes/MediaHandler.php [hunk 1]
```diff
@@ -316,6 +316,18 @@ protected function storeFile($fileData, $disk): Media
...
 
+            $image_watermart_path = config('nova-media-field.image_watermark_path', null);
+            if(!is_null($image_watermart_path)) {
+                //Save image without watermark
+                $rawFilePath = $storagePath . pathinfo($newFilename, PATHINFO_FILENAME) . "-RAW." . $origExtension;
+                $file = $image->encode($origExtension, config('nova-media-field.quality', 80));
+                $disk->put($rawFilePath, $file);
+
+                // Add watermark to image
+                $watermark = Image::make($image_watermart_path);
```
Comment (Tarpsvo) [inline src/Classes/MediaHandler.php]: Shouldn't this be after the original file is saved in the original flow at line 342? See: https://github.com/optimistdigital/nova-media-field/blob/dbc381af185600f545f5cc487e72e61456762b0c/src/Classes/MediaHandler.php#L340-L344

Your code seems to compress and save the images twice.

File: src/Classes/MediaHandler.php [hunk 2]
```diff
@@ -316,6 +316,18 @@ protected function storeFile($fileData, $disk): Media
             $origFile = file_get_contents($tmpPath . $tmpName);
             $image = Image::make($origFile);
 
+            $image_watermart_path = config('nova-media-field.image_watermark_path', null);
+            if(!is_null($image_watermart_path)) {
```
Comment (Tarpsvo) [inline src/Classes/MediaHandler.php]: This check should move to line 342 as well.

**Review by Tarpsvo — CHANGES_REQUESTED** [review]
Hi @pitchayakit! Thanks a lot for the suggestion and PR. :)

I looked over your code and brought up some questions. The main issue is that your flow currently saves and compresses the image twice. If you could address that, we'd be happy to merge the PR.

**Comment by pitchayakit** [comment]
I'm looking forward to your feedback. 
For the last step, I'm happy to update readme as well  :)

**Comment by Tarpsvo** [comment]
Thanks for the PR! I released it in version 2.1.0. :)

**Pull request merged**